### PR TITLE
perf: cleanup loop - extra ? checks

### PR DESCRIPTION
**Pull request by chiefcll**

### CODE REVIEW

File: src/core/Stage.ts [hunk 1]
```diff
@@ -387,18 +387,13 @@ export class Stage {
       node.renderQuads(this.renderer);
     }
 
-    for (let i = 0; i < node.children.length; i++) {
-      const child = node.children[i];
+    const childrenLength = node.children.length;
+    for (let i = 0; i < childrenLength; i++) {
+      const child = node.children[i] as CoreNode;
 
-      if (!child) {
```
Comment (elsassph) [inline src/core/Stage.ts:393]: I think `if (child)` isn't a luxury

Comment (chiefcll) [reply src/core/Stage.ts:393]: That was only added for typescript. You shouldn't have a child which doesn't exist

File: src/core/Stage.ts [hunk 2]
```diff
@@ -387,18 +387,13 @@ export class Stage {
...
-        continue;
+      if (child.worldAlpha !== 0) {
+        this.addQuads(child);
```
Comment (wouterlucas) [inline src/core/Stage.ts:395]: I'd prefer we keep it `=== 0` as a "off case" and keep the normal path down below as it was, just reads better in my opinion.

File: src/core/Stage.ts (hunk 1 above)
Comment (wouterlucas) [reply src/core/Stage.ts:393]: it can if the array was modified while you started the loop?

**Pull request closed without merging**